src/api/ucc.h

```diff
@@ -473,6 +473,20 @@ ucc_status_t ucc_lib_config_modify(ucc_lib_config_h *config, const char *name,
                                    const char *value);
 
 
+/**
+ *  @ingroup UCC_LIB
+ *
+ *  @brief The @ref ucc_init_version is an internal routine that checks
+ *  compatibility with particular UCC API version.
+ *  @ref ucc_init should be used to create UCC library handle.
+ */
+ucc_status_t ucc_init_version(unsigned api_major_version,
```

Comment:
@Sergei-Lebedev does it compile fine with ucc_init_version undefined? Maybe add src/core/ucc_lib.c and put ucc_init_version with empty implementation there?

Reply:
Is there a need to expose in ucc.h ? If it is purely internal, I prefer not to expose in api/ucc.h.

Reply:
@vspetrov 
Fixed
@manjugv 
Yes, that is the whole idea, because when user will compile the application this application will automatically contain ucc api version. If then the application is ran with wrong ucc lib it will report error.
Same is done in UCP https://github.com/openucx/ucx/blob/5c6bd6ed6d788df6ddcbab9ecd2c662bf3fae85b/src/ucp/api/ucp.h#L1586

Reply:
Thanks for the context. 

Can you define a different Doxygen group for this, so, we will not generate it as a part of the specification? May be `@ingroup UCC_LIB_INTERNAL` or something ? 

"particular UCC API version" > particular **a** UCC API version
to create UCC library handle > to create **the** UCC library handle

Reply:
@manjugv fixed